torch/_refs/__init__.py

```diff
@@ -3752,31 +3726,22 @@ def arange(
     pin_memory: bool = False,
     requires_grad: bool = False,
 ) -> TensorLikeType:
-    assert (a is not None and b is not None) or (a is not None and b is None)
-    if a is not None and b is not None:
-        return prims.arange(
-            a,
-            b,
-            step,
-            dtype=dtype,
-            device=device,
-            # layout=layout,
-            # pin_memory=pin_memory,
-            requires_grad=requires_grad,
-        )
-    elif a is not None and b is None:
-        return prims.arange(
-            0,
-            a,
-            step,
-            dtype=dtype,
-            device=device,
-            # layout=layout,
-            # pin_memory=pin_memory,
-            requires_grad=requires_grad,
-        )
-    else:
-        raise AssertionError()
+    assert not pin_memory
```

Comment:
`assert` -> `utils.check`

Reply:
I just wrote it as an assert as it's more of a hard fail as it's not implemented. I could use a `utils.check` with a `NotImplmentedError`, but I think shorter is better here.

Reply:
Addressed here https://github.com/pytorch/pytorch/pull/85333